ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1187 @@
+            );
+        }
+    }
+}
+
+// Add language support.
+extend(
+    defaultOptions.lang,
+    /**
+     * @optionparent lang
+     */
+    {
+        /**
+         * @since    next
+         * @product  highcharts
+         */
+        breadcrumbsToLabel: '&#8592',
+
+        /**
+         * @since    next
+         * @product  highcharts
+         */
+        mainBreadcrumb: 'Main'
+    }
+);
+
+/**
+ * The Breadcrumbs class
+ *
+ * @class
+ * @name Highcharts.Breadcrumbs
+ *
+ * @param {Highcharts.Chart} chart
+ *        Chart object
+ * @param {Highcharts.Options} userOptions
+ *        User options
+ */
+class Breadcrumbs {
+    /**
+     * Options for breadcrumbs
+     *
+     * @since   next
+     * @product highcharts
+     */
+    public static defaultBreadcrumbsOptions = {
+        /**
+         * Them for button.
+         *
+         * @type {Highcharts.SVGAttributes}
+         */
+        buttonTheme: {
+            /** @ignore */
+            height: 18,
+            /** @ignore */
+            padding: 2,
+            /** @ignore */
+            zIndex: 7
+        },
+
+        /**
+         * The default padding for each button and separator in each direction.
+         *
+         * @type      {number}
+         * @since     next
+         */
+        buttonSpacing: 5,
+
+        /**
+         * Fires when clicking on the breadcrumbs button.
+         * Two arguments are passed to the function. First breadcrumb button
+         * as an SVG element. Second is the breadcrumbs class,
+         * containing reference to the chart, series etc.
+         *
+         * ```js
+         * click: function(button, breadcrumbs) {
+         *   console.log(button);
+         * }
+         * ```
+         *
+         * Return false to stop default buttons click action.
+         *
+         * @type      {Highcharts.BreadcrumbsClickCallbackFunction}
+         * @since     next
+         * @apioption breadcrumbs.events.click
+         */
+
+        /**
+         * When the breadcrumbs are floating, the plot area will not move
+         * to make space for it. By default, the chart will not make space
+         * for the buttons. This property won't work when positioned in the
+         * middle.
+         *
+         * @sample {highcharts} highcharts/breadcrumbs/floating-false
+         *          Non-floating breadcrumbs
+         *
+         * @type      {boolean}
+         * @since     next
+         */
+        floating: false,
+
+        /**
+         * A format string for the breadcrumbs button.
+         * Variables are enclosed by curly brackets.
+         * Available values are passed in the declared point options.
+         *
+         * @type      {string|undefined}
+         * @since     next
+         * @default   undefined
+         * @sample {highcharts} highcharts/breadcrumbs/format
+         *          Display custom values in breadcrumb button.
+         */
+        format: void 0 as any,
```

Comment:
Unnecessary any casts. Undefined is a valid type.

Reply:
corrected